packages/core/src/utils.ts

```diff
@@ -464,6 +464,24 @@ export function isEmpty(v: any): boolean {
   return (t === 'string' && v === '') || (t === 'object' && Object.keys(v).length === 0);
 }
 
+type CanBePopulated = { length: number } | object;
+/**
+ * Returns true if the value is a non-empty string, an object with a length property greater than zero, or a non-empty object
+ * @param arg - Any value
+ * @returns True if the value is a non-empty string, an object with a length property greater than zero, or a non-empty object
+ */
+export function isPopulated<T extends { length: number } | object>(arg: CanBePopulated | undefined | null): arg is T {
```

Comment:
I've been wanting this function to exist for a while; the type-narrowing is the really nice part here.

It would have been nice to redefine `isEmpty(arg)` to just be `!isPopulated(arg)`, but they have slightly different semantics.

I'm in favor of doing the work to make `isEmpty` the inverse of `isPopulated`. Or at least deprecating `isEmpty` for now in favor of using `isPopulated`

Reply:
Neat.  I'd be curious to review the cases where `isEmpty(arg) != !isPopulated(arg)`.  `isEmpty()` has a lot of weird edge cases to satisfy FHIRPath test cases.  If there's a good clean conceptual model that fulfills it, that'd be perfect.

Reply:
I made a few tweaks to where they're logically consistent now except for numerical, boolean, etc. are expected to return `false` for both `isEmpty` and `isPopulated`.